src/Http/Http/src/SendFileFallback.cs

```diff
@@ -20,6 +23,37 @@ public static class SendFileFallback
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> used to abort the transmission.</param>
         /// <returns></returns>
         public static async Task SendFileAsync(Stream destination, string filePath, long offset, long? count, CancellationToken cancellationToken)
+        {
+            FileStream fileStream = GetFileStream(filePath, offset, count, DefaultBufferSize, cancellationToken);
+
+            using (fileStream)
+            {
+                fileStream.Seek(offset, SeekOrigin.Begin);
+                await StreamCopyOperationInternal.CopyToAsync(fileStream, destination, count, DefaultBufferSize, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Write the segment of the file using pipe writer.
+        /// </summary>
+        /// <param name="writer"></param>
+        /// <param name="filePath">The full disk path to the file.</param>
+        /// <param name="offset">The offset in the file to start at.</param>
+        /// <param name="count">The number of bytes to send, or null to send the remainder of the file.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> used to abort the transmission.</param>
+        /// <returns></returns>
+        public static async Task SendFileAsync(PipeWriter writer, string filePath, long offset, long? count, CancellationToken cancellationToken)
+        {
+            FileStream fileStream = GetFileStream(filePath, offset, count, DefaultBufferSize, cancellationToken);
```

Comment:
`await using`

Reply:
I added the `using` without curly brackets, but I'm not sure what to do with `await`.

Reply:
That's the DisposeAsync synatx:
https://stackoverflow.com/questions/58696427/proper-way-to-use-disposeasync-on-c-sharp-streams

Reply:
Thanks. I forgot about that.